### ISSUE TITLE
Running Glances webserver as service using NSSM not working with --password

### ISSUE THREAD

**Issue by GlobalRisk** [nicolargo/glances#1381]
#### Description

I'm trying to run the glances webserver as a service using NSSM and then use a reverse proxy to connect to it. It all worked well so I tried to get some authentication going by using the --password command but then glances won't run anymore (the service doesn't give any error and it just starts like usually but I can't connect to the webserver anymore). I suspect it just refuses to run because I don't get any logs.

I know I'm doing it right because when I start glances from the cmd it works perfectly... So this seems to be an issue with NSSM. Is there perhaps a workaround or some other way to run the webserver as a service with some authentication? Also is there a way to change the default username?

#### Versions

* Glances & psutil (glances -V): Glances v3.0.2 with psutil v5.4.8
* Operating System (lsb_release -a): Microsoft Windows 10.0.17134

**Comment by nicolargo** [comment]
Hi @GlobalRisk 

perhaps, using NSSM Glances can not found the glances.pwd file where the log/password hash is saved the first time you run Glances. This file should be located in %APPDATA%\glances and should be named glances.pwd.

You can also try to run Glances in verbose/debug mode (-d) in order to have additional log file in the glances-<user>.log file.

Keep us inform.

**Comment by GlobalRisk** [comment]
I gave up on running the webserver as a service and just used task scheduler. Is there any way that I can change the default username? Thanks for the help btw :)

**Comment by nicolargo** [comment]
After deleting the glances.pwd file, run Glances with the following command
line:

    glances --username --password

**Comment by GlobalRisk** [comment]
Yeah I read that in the docs but how to make it so that it defaults to that file and doens't ask me for a username the next time I run it with these arguments?

**Comment by nicolargo** [comment]
Correct ! It miss an option to set the username when the login/password has already been created.

I will implement it in the DEVELOP branch as soon as possible.
The behavior **will be** the following:

First of all create a login/password couple:

    glances -w --username --password

If you save the login/password, it will create a <username>.pwd file in the same folder than the glances.conf file.

After you will have to run Glances with (a non existing) option -u:

    glances -w -u <username>

Thanks for the head up.

**Comment by GlobalRisk** [comment]
That's great to hear :) Any idea what could cause the bug?

**Comment by nicolargo** [comment]
Done in the DEVELOP branch.

Can you test it ?

**Comment by GlobalRisk** [comment]
Sure, give me a sec :)

**Comment by GlobalRisk** [comment]
Ah nvm, my bad. I had forgotten the -w argument and that's when it started on 61209 😅 
Thanks anyways 👍
